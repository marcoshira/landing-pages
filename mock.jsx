import linksMock from '../../components/NavLinks/mock';
import gridMock from '../../components/GridText/mock';
import { GridText } from '../../components/GridText';

export const mockBase = {
  children: (
    <>
      <GridText {...gridMock} background={'navy'} />
      <GridText {...gridMock} />
      <GridText {...gridMock} background={'navy'} />
      <GridText {...gridMock} />
      <GridText {...gridMock} background={'navy'} />
      <GridText {...gridMock} />
    </>
  ),
  links: linksMock,
  logoData: {
    text: 'Logo',
    link: '#',
  },
  footerHtml: '<p>Teste de footer</p>',
};
