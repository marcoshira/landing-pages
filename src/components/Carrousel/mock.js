export default {
  background: 'white',
  color: 'navy',
  title: 'My grid',
  description: 'Atue doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  flex: [
    {
      altText: 'Teste 1',
      srcImg:
        'https://images.unsplash.com/photo-1643833728695-b3923431306d?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDY2Ng&ixlib=rb-1.2.1&q=80&w=800',
    },
    {
      altText: 'Teste 2',
      srcImg:
        'https://images.unsplash.com/photo-1644027616320-b378fc57f78e?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDcxOQ&ixlib=rb-1.2.1&q=80&w=800',
    },
    {
      altText: 'Teste 3',
      srcImg:
        'https://images.unsplash.com/photo-1643407997122-86f5534dcbd8?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDc0Ng&ixlib=rb-1.2.1&q=80&w=800',
    },
    {
      altText: 'Teste 4',
      srcImg:
        'https://images.unsplash.com/photo-1644131575146-2312c50c97e9?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDc3OQ&ixlib=rb-1.2.1&q=80&w=800',
    },
  ],
};
