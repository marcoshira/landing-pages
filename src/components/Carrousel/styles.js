import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Flex = styled.div`
  ${({ theme }) => css`
    overflow: hidden;

    width: 61.75%;
    height: 50rem;
    white-space: nowrap;
    position: relative;
    /* border: 8px solid red; */
    left: 19.3%;
    padding: 0;

    @media ${theme.media.lteMedium} {
      width: 100%;
      left: 0%;
    }
  `}
`;

export const FlexElement = styled.div`
  ${({ theme, imageCounter }) => css`
    display: flex;
    flex-direction: column;
    overflow: hidden;
    width: 100%;
    justify-content: center;
    align-items: center;
    overflow: hidden;
    opacity: 0;
    visibility: hidden;
    position: absolute;
    transition: all 300ms linear;

    :nth-of-type(${imageCounter}) {
      visibility: visible;
      opacity: 1;
      position: static;
      border: none;
      /* display: none; */
    }
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;
    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;

export const ButtonLeft = styled.button`
  ${({ theme }) => css`
    z-index: 4;
    /* position: fixed; */
    width: 4rem;
    height: 4rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: 0.1rem solid ${theme.colors.mediumGray};
    opacity: 75%;

    > svg {
      width: 2.5rem;
      height: 2.5rem;
    }
    /*
    position: absolute;
    display: inline; */
    position: relative;
    top: 27rem;
    left: calc(15% -4rem);

    /* @media ${theme.media.lteMedium} {
      top: 45rem;
    } */
    @media ${theme.media.lteSmall} {
      top: 20rem;
      left: -5%;
    }
  `}
`;

export const ButtonRight = styled.button`
  ${({ theme }) => css`
    z-index: 4;
    /* position: fixed; */
    width: 4rem;
    height: 4rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: 0.1rem solid ${theme.colors.mediumGray};
    opacity: 75%;

    > svg {
      width: 2.5rem;
      height: 2.5rem;
    }
    position: relative;
    top: 27rem;
    left: calc(95% - 3vw);

    @media ${theme.media.lteMedium} {
      left: 80%;
    }

    @media ${theme.media.lteSmall} {
      top: 20rem;
      left: 82%;
    }
  `}
`;
