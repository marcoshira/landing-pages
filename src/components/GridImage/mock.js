export default {
  background: false,
  title: 'My grid',
  description: 'Atue doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      altText: 'Teste 1',
      srcImg:
        'https://images.unsplash.com/photo-1643833728695-b3923431306d?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDY2Ng&ixlib=rb-1.2.1&q=80&w=800',
    },
    {
      altText: 'Teste 2',
      srcImg:
        'https://images.unsplash.com/photo-1644027616320-b378fc57f78e?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDcxOQ&ixlib=rb-1.2.1&q=80&w=800',
    },
    {
      altText: 'Teste 3',
      srcImg:
        'https://images.unsplash.com/photo-1643407997122-86f5534dcbd8?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDc0Ng&ixlib=rb-1.2.1&q=80&w=800',
    },
    {
      altText: 'Teste 4',
      srcImg:
        'https://images.unsplash.com/photo-1644131575146-2312c50c97e9?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDc3OQ&ixlib=rb-1.2.1&q=80&w=800',
    },
    {
      altText: 'Teste 5',
      srcImg:
        'https://images.unsplash.com/photo-1643205942704-b6bb6d3115f1?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDc5NQ&ixlib=rb-1.2.1&q=80&w=800',
    },
    {
      altText: 'Teste 6',
      srcImg:
        'https://images.unsplash.com/photo-1640597995884-57667d173ccf?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDgwOQ&ixlib=rb-1.2.1&q=80&w=800',
    },
  ],
};
