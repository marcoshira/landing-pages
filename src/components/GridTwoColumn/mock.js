export default {
  title: 'Grid two columns',
  text: `Lore, ipsum dolor sit amet consectetur adipisicing elit. Atque a, nostrum architecto modi omnis adipisci, est ullam in fugit expedita sapiente aperiam et quasi facere quisquam consequuntur numquam perspiciatis laudantium?`,
  srcImg: 'assets/images/javascript.svg',
};
