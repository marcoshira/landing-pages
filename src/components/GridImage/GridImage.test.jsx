import { screen } from '@testing-library/react';
import { GridImage } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';
import mock from './mock';

describe('<GridImage />', () => {
  it('should render', () => {
    const { container } = renderTheme(<GridImage {...mock} />);
    expect(container).toMatchSnapshot();
  });

  it('should render without background and text color', () => {
    const { container } = renderTheme(<GridImage {...mock} background={undefined} color={undefined} />);
    expect(container).toMatchSnapshot();
  });
});
