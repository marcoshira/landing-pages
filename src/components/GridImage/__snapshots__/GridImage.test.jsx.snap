// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImage /> should render 1`] = `
.c0 {
  background: #FFFFFF;
  color: #0A1128;
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  color: inherit;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-bottom: 6.4rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  -webkit-transform: scale(1.2) rotate(10deg);
  -ms-transform: scale(1.2) rotate(10deg);
  transform: scale(1.2) rotate(10deg);
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
    color="navy"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          My grid
        </h2>
        <p
          class="c4 c5"
        >
          Atue doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </p>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="Teste 1"
              class="c8"
              src="https://images.unsplash.com/photo-1643833728695-b3923431306d?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDY2Ng&ixlib=rb-1.2.1&q=80&w=800"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 2"
              class="c8"
              src="https://images.unsplash.com/photo-1644027616320-b378fc57f78e?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDcxOQ&ixlib=rb-1.2.1&q=80&w=800"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 3"
              class="c8"
              src="https://images.unsplash.com/photo-1643407997122-86f5534dcbd8?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDc0Ng&ixlib=rb-1.2.1&q=80&w=800"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 4"
              class="c8"
              src="https://images.unsplash.com/photo-1644131575146-2312c50c97e9?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDc3OQ&ixlib=rb-1.2.1&q=80&w=800"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 5"
              class="c8"
              src="https://images.unsplash.com/photo-1643205942704-b6bb6d3115f1?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDc5NQ&ixlib=rb-1.2.1&q=80&w=800"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 6"
              class="c8"
              src="https://images.unsplash.com/photo-1640597995884-57667d173ccf?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDgwOQ&ixlib=rb-1.2.1&q=80&w=800"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridImage /> should render without background and text color 1`] = `
.c0 {
  background: #FFFFFF;
  color: #0A1128;
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  color: inherit;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-bottom: 6.4rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  -webkit-transform: scale(1.2) rotate(10deg);
  -ms-transform: scale(1.2) rotate(10deg);
  transform: scale(1.2) rotate(10deg);
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
    color="navy"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          My grid
        </h2>
        <p
          class="c4 c5"
        >
          Atue doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </p>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="Teste 1"
              class="c8"
              src="https://images.unsplash.com/photo-1643833728695-b3923431306d?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDY2Ng&ixlib=rb-1.2.1&q=80&w=800"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 2"
              class="c8"
              src="https://images.unsplash.com/photo-1644027616320-b378fc57f78e?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDcxOQ&ixlib=rb-1.2.1&q=80&w=800"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 3"
              class="c8"
              src="https://images.unsplash.com/photo-1643407997122-86f5534dcbd8?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDc0Ng&ixlib=rb-1.2.1&q=80&w=800"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 4"
              class="c8"
              src="https://images.unsplash.com/photo-1644131575146-2312c50c97e9?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDc3OQ&ixlib=rb-1.2.1&q=80&w=800"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 5"
              class="c8"
              src="https://images.unsplash.com/photo-1643205942704-b6bb6d3115f1?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDc5NQ&ixlib=rb-1.2.1&q=80&w=800"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 6"
              class="c8"
              src="https://images.unsplash.com/photo-1640597995884-57667d173ccf?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=800&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY0NTIwMDgwOQ&ixlib=rb-1.2.1&q=80&w=800"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
